src/app/page.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import Image from "next/image";
+import React, { useState } from "react";
+import SingleFileUpload from "../components/singleFile";
+
+export default function App() {
+  const [fileData, setPreviewFileData] = useState(
+    {} as {
+      previewType: string;
+      previewUrl: string | ArrayBuffer | null;
+      previewName: string;
+      isDragging: boolean;
+      isLoading: boolean;
+    }
+  );
+
+  const getPreviewFileData = (file: any) => {
+    setPreviewFileData(file);
+  };
+
+  const [uploadingStatus, setUploadingData] = useState(false);
+
+  const isUploading = (flag: boolean) => {
+    setUploadingData(flag);
+
+    if (flag) {
+      setTimeout(() => {
+        setUploadingData(!flag);
+      }, 5000);
+    }
+  };
+
+  const [getFile, setFileData] = useState({});
+
+  const getFileData = (file: any) => {
+    setFileData(file);
+  };
+
+  return (
+    <main className="flex min-h-screen flex-col justify-between p-5">
+      <SingleFileUpload
+        getPreview={getPreviewFileData}
+        getFileObj={getFileData}
+        isUploading={isUploading}
+        uploadingStatus={uploadingStatus}
+        uploadBtn={"Save"}
+        progressBtn={"Saving..."}
+      >
+        <div className="m-5">
+          <div className="flex items-center justify-center">
+            {!fileData || !fileData.previewUrl ? (
+              <label className="flex flex-col items-center justify-center w-full h-56 border-2 border-gray-300 border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600">
```

Comment:
Let's use [this](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier), and sort classes for better readability